From 2daaf2d60e0b0b3dacd21022dca8acf78be4e973 Mon Sep 17 00:00:00 2001
From: Fabian Deutsch <fabian.deutsch@gmx.de>
Date: Wed, 6 Apr 2011 14:56:17 +0200
Subject: [PATCH 2/2] Add LinkMonitor-Tweak.

This tweak allows to link or unlink monitors in a more-than-one-monitor setup.

Signed-off-by: Fabian Deutsch <fabian.deutsch@gmx.de>
---
 gtweak/tweaks/tweak_shell.py |    9 ++++++++-
 1 files changed, 8 insertions(+), 1 deletions(-)

diff --git a/gtweak/tweaks/tweak_shell.py b/gtweak/tweaks/tweak_shell.py
index fb8d359..19753b2 100644
--- a/gtweak/tweaks/tweak_shell.py
+++ b/gtweak/tweaks/tweak_shell.py
@@ -26,7 +26,7 @@ from gi.repository import GLib
 from gtweak.gsettings import GSettingsSetting
 from gtweak.gshellwrapper import GnomeShell
 from gtweak.tweakmodel import Tweak, TweakGroup
-from gtweak.widgets import GConfComboTweak, GSettingsComboEnumTweak, GSettingsSwitchTweak, build_label_beside_widget, build_horizontal_sizegroup
+from gtweak.widgets import GConfSwitchTweak, GConfComboTweak, GSettingsComboEnumTweak, GSettingsSwitchTweak, build_label_beside_widget, build_horizontal_sizegroup
 
 class ShowWindowButtons(GConfComboTweak):
     def __init__(self, **options):
@@ -39,6 +39,12 @@ class ShowWindowButtons(GConfComboTweak):
             (':minimize,maximize,close', 'All')),
             **options)
 
+class LinkMonitors(GConfSwitchTweak):
+    def __init__(self, **options):
+        GConfSwitchTweak.__init__(self,
+            "/desktop/gnome/shell/windows/workspaces_only_on_primary",
+            **options)
+
 class _ThemeZipChooser(Gtk.FileChooserButton):
     def __init__(self):
         Gtk.FileChooserButton.__init__(self, title="Select theme file")
@@ -159,6 +165,7 @@ TWEAK_GROUPS = (
             GSettingsSwitchTweak("org.gnome.shell.clock", "show-date", schema_filename="org.gnome.shell.gschema.xml"),
             ShowWindowButtons(size_group=sg),
             ThemeInstaller(size_group=sg),
+            LinkMonitors(size_group=sg),
             GSettingsComboEnumTweak("org.gnome.settings-daemon.plugins.power", "lid-close-battery-action", size_group=sg),
             GSettingsComboEnumTweak("org.gnome.settings-daemon.plugins.power", "lid-close-ac-action", size_group=sg)),
 )
-- 
1.7.4.2

